src/Microsoft.Identity.Web/Resource/RegisterValidAudience.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.Identity.Web.Resource
+{
+    /// <summary>
+    /// Generic class that registers the token audience from the provided Azure AD authority.
+    /// </summary>
+    internal class RegisterValidAudience
+    {
+        private string ClientId { get; set; }
+        private bool IsB2C { get; set; } = false;
+        private const string Version = "ver";
+        private const string V1 = "1.0";
+        private const string V2 = "2.0";
+
+        public void RegisterAudienceValidation(
+            TokenValidationParameters validationParameters,
+            MicrosoftIdentityOptions microsoftIdentityOptions)
+        {
+            if (validationParameters == null)
+            {
+                throw new ArgumentNullException(nameof(validationParameters));
+            }
+
+            ClientId = microsoftIdentityOptions.ClientId;
+            IsB2C = microsoftIdentityOptions.IsB2C;
+
+            validationParameters.AudienceValidator = ValidateAudience;
+        }
+
+        internal /*for test only*/ bool ValidateAudience(
+            IEnumerable<string> audiences,
+            SecurityToken securityToken,
+            TokenValidationParameters validationParameters)
+        {
+            JwtSecurityToken token = securityToken as JwtSecurityToken;
+            string validAudience;
```

Comment:
Do we need this? Can we just pass string to `Contains`?

Reply:
I don't know what you mean.

---
In reply to: [425456868](https://github.com/AzureAD/microsoft-identity-web/pull/163#discussion_r425456868) [](ancestors = 425456868)

Reply:
Meant like so in the below code:
```csharp
return audiences.Contains($"{ClientId}");

return audiences.Contains($"api://{ClientId}");
```